Tools are not available on cross compiled platforms since #61861

Moved from https://github.com/rust-lang/rust-clippy/issues/4267

#61861 was merged in #61962, since then Rustfmt fails to build when cross compiling which is the case for almost all non Tier 1 platforms.
Clippy and RLS are failing with similar error.

Error on `aarch64-unknown-linux-gnu`:
```
[01:14:03] error[E0464]: multiple matching crates for `rustc_macros`
[01:14:03]   --> /cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-491.0.0/abi/mod.rs:10:5
[01:14:03]    |
[01:14:03]    | use syntax_pos::symbol::{sym, Symbol};
[01:14:03]    |     ^^^^^^^^^^
[01:14:03]    |
[01:14:03]    = note: candidates:
[01:14:03]            crate `rustc_macros`: /checkout/obj/build/x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_macros-97bf1e9c420f9886.so
[01:14:03]            crate `rustc_macros`: /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/release/deps/librustc_macros-2f4546e1163449fb.so
[01:14:03] 
[01:14:03] error[E0461]: couldn't find crate `rustc_macros` with expected target triple x86_64-unknown-linux-gnu which `syntax_pos` depends on
[01:14:03]   --> /cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-491.0.0/abi/mod.rs:10:5
[01:14:03]    |
[01:14:03] 10 | use syntax_pos::symbol::{sym, Symbol};
[01:14:03]    |     ^^^^^^^^^^
[01:14:03]    |
[01:14:03]    = note: the following crate versions were found:
[01:14:03]            crate `rustc_macros`, target triple aarch64-unknown-linux-gnu: /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/aarch64-unknown-linux-gnu/release/deps/librustc_macros-55389246a2ce204b.so
[01:14:03] 
[01:14:03] error: aborting due to 2 previous errors
```
https://api.travis-ci.com/v3/job/209363492/log.txt

@rustbot modify labels: T-dev-tools A-rustbuild

Also clippy is failing with another error:
```
Building stage1 tool clippy-driver (armv7-unknown-linux-gnueabihf)
   Compiling cfg-if v0.1.8
   Compiling matches v0.1.8
   Compiling rustc-demangle v0.1.16
   Compiling unicode-normalization v0.1.7
   Compiling unicode-width v0.1.6
   Compiling pulldown-cmark v0.5.3
   Compiling itoa v0.4.4
   Compiling percent-encoding v2.0.0
   Compiling lazy_static v1.3.0
   Compiling ucd-util v0.1.3
   Compiling rustc_tools_util v0.2.0 (/builds/nsidious/rust-ci/rust/src/tools/clippy/rustc_tools_util)
   Compiling quine-mc_cluskey v0.2.4
   Compiling smallvec v0.6.10
   Compiling utf8-ranges v1.0.2
   Compiling if_chain v1.0.0
   Compiling libc v0.2.62
   Compiling ryu v1.0.0
   Compiling unicode-bidi v0.3.4
   Compiling memchr v2.2.0
   Compiling bitflags v1.1.0
   Compiling getopts v0.2.21
   Compiling thread_local v0.3.6
   Compiling regex-syntax v0.6.6
   Compiling regex v1.1.6
   Compiling clippy v0.0.212 (/builds/nsidious/rust-ci/rust/src/tools/clippy)
   Compiling backtrace-sys v0.1.30
   Compiling unicase v2.4.0
   Compiling idna v0.2.0
   Compiling aho-corasick v0.7.3
   Compiling backtrace v0.3.37
   Compiling serde_json v1.0.40
   Compiling toml v0.5.3
   Compiling url v2.1.0
   Compiling failure v0.1.5
   Compiling cargo_metadata v0.8.0
   Compiling clippy_lints v0.0.212 (/builds/nsidious/rust-ci/rust/src/tools/clippy/clippy_lints)
error[E0463]: can't find crate for `fmt_macros`
  --> src/tools/clippy/clippy_lints/src/lib.rs:20:1
   |
20 | extern crate fmt_macros;
   | ^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: Could not compile `clippy_lints`.

To learn more, run the command again with --verbose.
```

--- a/src/main.rs
+++ b/src/main.rs
@@ -308,14 +308,8 @@ fn publish(pkg: &Package, commit: &str, vers: &semver::Version) {
                             "package".to_string(),
                             toml::Value::String(format!("{}-{}", PREFIX, name)),
                         );
-                        has_package = true;
                     }
-                    let key_name = if has_package {
-                        name.clone()
-                    } else {
-                        format!("{}-{}", PREFIX, name)
-                    };
-                    (key_name, new_table.into())
+                    (name.clone(), new_table.into())
                 })
                 .collect::<Vec<_>>();
             toml.insert(

--- a/src/main.rs
+++ b/src/main.rs
@@ -303,6 +303,13 @@ fn publish(pkg: &Package, commit: &str, vers: &semver::Version) {
                         }
                     }
                     new_table.insert("version".to_string(), toml::Value::String(vers.to_string()));
+                    if !has_package {
+                        new_table.insert(
+                            "package".to_string(),
+                            toml::Value::String(format!("{}-{}", PREFIX, name)),
+                        );
+                        has_package = true;
+                    }
                     let key_name = if has_package {
                         name.clone()
                     } else {

